app-crypt/tpm2-tools/tpm2-tools-5.2.ebuild

```diff
@@ -0,0 +1,71 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{8,9} )
+inherit autotools bash-completion-r1 python-single-r1
+
+DESCRIPTION="Tools for the TPM 2.0 TSS"
+HOMEPAGE="https://github.com/tpm2-software/tpm2-tools"
+SRC_URI="https://github.com/tpm2-software/${PN}/releases/download/${PV}/${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="+fapi test"
+
+RDEPEND=">=app-crypt/tpm2-tss-3.0.1[fapi?]
+	dev-libs/openssl:=
+	net-misc/curl
+	sys-libs/efivar:=
+	${PYTHON_DEPS}"
+DEPEND="${RDEPEND}
+	test? (
+		app-crypt/swtpm
+		app-crypt/tpm2-abrmd
+		app-editors/vim-core
+		dev-tcltk/expect
+		dev-util/cmocka
+		dev-python/pyyaml
+	)"
+BDEPEND="virtual/pkgconfig
+	sys-devel/autoconf-archive"
```

Comment:
`${PYTHON_DEPS}` should be here as well (executed at build time by python-single-r1, and things in non-B DEPEND can't be executed when cross-compiling).

For the same reason, believe /at least/ pyyaml, expect, and vim-core should be moved in BDEPEND as well (realistically we don't run tests when cross-compiling, but it doesn't hurt to put deps where they belong). Not sure for the others except cmocka (cmocka is linked with rather than executed) without looking deeper.

Reply:
I'm a little unclear on BDEPEND. The way I understand it, stuff in BDEPEND is runs on $CBUILD and DEPEND/RDEPEND on $CHOST, however tests aren't run on cross compile, so should testing-only stuff always be on BDEPEND ? I'm not so sure now...

Reply:
Well there are cases where you /could/ run tests when cross-compiling (term used loosely), it's often possible with scripts, e.g. python/perl would have `BDEPEND="test? ( ${RDEPEND} )"` (typically handled by eclasses) to ensure their runtime deps are available for testing on CBUILD.

And then things like xxd (vim-core) would be executed and belong in CBUILD.

But as I said, this is more about pedantic and having dependencies in the right block, in cases where there's compiled code we won't be able to run tests anyway (so if you'd rather not bother, can leave this alone).